src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -1721,6 +1721,11 @@ private void ReadExpandedNestedResourceInfoEnd(bool isCollection)
                 else
                 {
                     // null resource
+                    if (this.CurrentResourceTypeReference.IsODataComplexTypeKind() || this.CurrentResourceTypeReference.IsODataUnTypedTypeKind())

```

Comment:
>IsODataComplexTypeKind [](start = 58, length = 22)

shall we add "Enum" check?

Reply:
Collection of primitives and enums do not go through this check.

---
In reply to: [266678953](https://github.com/OData/odata.net/pull/1322#discussion_r266678953) [](ancestors = 266678953)